dist/tools/doccheck/check.sh

```diff
@@ -10,6 +10,11 @@
 
 RIOTBASE="$(cd $(dirname $0)/../../..; pwd)"
 
+die() {
+   echo "warning: Failed to build run doxygen"
```

Comment:
these error messages can come from the Doxygen makefile. E.g there might be a "doxygen not found" or a "git-lfs" not installed, they are 2 different errors

Reply:
The idea of this PR was to catch any non-zero exit code from `make doc`. Maybe I can change the message to "make doc exited with non-zero code" to make it explicit.
Afterwards we could add filters to point at the cause of the problem.